EventFilter/HcalRawToDigi/plugins/HcalDigiToRawuHTR.cc

```diff
@@ -0,0 +1,285 @@
+  // -----------------------------------------------------
+  // loop over each uHTR and format data
+  // -----------------------------------------------------
+  // -----------------------------------------------------
+  // loop over each uHTR and format data
+  int idxuhtr =-1;
+  for( UHTRpacker::UHTRMap::iterator uhtr = uhtrs.uhtrs.begin() ; uhtr != uhtrs.uhtrs.end() ; ++uhtr){
+
+    idxuhtr ++;
+   
+    uint64_t crateId = (uhtr->first)&0xFF;
+    uint64_t slotId =  (uhtr->first&0xF00)>>8;
+
+    uhtrs.finalizeHeadTail(&(uhtr->second),_verbosity);
+    int fedId = FEDNumbering::MINHCALuTCAFEDID + crateId;
+    if( fedMap.find(fedId) == fedMap.end() ){
+      /* QUESTION: where should the orbit number come from? */
+      fedMap[fedId] = std::unique_ptr<HCalFED>(new HCalFED(fedId,iEvent.id().event(),iEvent.orbitNumber(),iEvent.bunchCrossing()));
+    }
+    fedMap[fedId]->addUHTR(uhtr->second,crateId,slotId);
+  }// end loop over uhtr containers
+
+  /* ------------------------------------------------------
+     ------------------------------------------------------
+           putting together the FEDRawDataCollection
+     ------------------------------------------------------
+     ------------------------------------------------------ */
+  for( map<int,unique_ptr<HCalFED> >::iterator fed = fedMap.begin() ; fed != fedMap.end() ; ++fed ){
+
+    int fedId = fed->first;
+
+    FEDRawData* rawData =  fed->second->formatFEDdata();
+    FEDRawData& fedRawData = fed_buffers->FEDData(fedId);
+    fedRawData = *rawData;
+    delete rawData;
+
+    FEDHeader hcalFEDHeader(fedRawData.data());
+    hcalFEDHeader.set(fedRawData.data(), 1, iEvent.id().event(), iEvent.bunchCrossing(), fedId);
+    FEDTrailer hcalFEDTrailer(fedRawData.data()+(fedRawData.size()-8));
+    hcalFEDTrailer.set(fedRawData.data()+(fedRawData.size()-8), fedRawData.size()/8, evf::compute_crc(fedRawData.data(),fedRawData.size()), 0, 0);
+
+  }// end loop over FEDs with data
+
+  iEvent.put(std::move(fed_buffers));
+  
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void HcalDigiToRawuHTR::fillDescriptions(edm::ConfigurationDescriptions& descriptions){
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.addUntracked<int>("Verbosity", 0);
+  desc.add<std::string>("ElectronicsMap", "");
+  desc.add<edm::InputTag>("QIE10", edm::InputTag("simHcalDigis", "HFQIE10DigiCollection"));
+  desc.add<edm::InputTag>("QIE11", edm::InputTag("simHcalDigis", "HBHEQIE11DigiCollection"));
+  desc.add<edm::InputTag>("HBHEqie8", edm::InputTag("simHcalDigis"));
+  desc.add<edm::InputTag>("HFqie8", edm::InputTag("simHcalDigis"));
+  desc.add<edm::InputTag>("TP", edm::InputTag("simHcalTriggerPrimitiveDigis"));
+  descriptions.addDefault(desc);
```

Comment:
Since we know we have to revert to using the QIE8 dataframe methods for eid lookup, can we bundle the changes to the description and the eid lookup changes into the next PR?